erpnext/projects/doctype/task/task.js

```diff
@@ -38,6 +38,16 @@ frappe.ui.form.on("Task", {
 				filters: filters
 			};
 		})
+
+		let prev_route = frappe.get_prev_route();
+		if (prev_route && prev_route[1] === 'Task') {
```

Comment:
@meshramsaroj Will this clear out the already existing fields in child table? please check. If it does, add a condition that the child table should be empty. something like
```suggestion
		if (prev_route && prev_route[1] === 'Task')  && (!frm.doc.projects){
```

Reply:
@sahil28297  fixed changes.
Please check